### PR TITLE
Fix rush add -p affecting all subspace projects

### PR DESCRIPTION
**Pull request by william2958**
## Summary
Currently running `rush add -p <package_name>` results in every project being updated in a subspace. This fixes the issue.

### CODE REVIEW

File: libraries/rush-lib/src/logic/PackageJsonUpdater.ts [hunk 1]
```diff
@@ -325,6 +325,7 @@ export class PackageJsonUpdater {
       this._rushConfiguration
     );
 
+    // console.log('adding project, updating projects: ', projects);
```
Comment (chengcyber) [inline libraries/rush-lib/src/logic/PackageJsonUpdater.ts]: remove this line?

File: common/changes/@microsoft/rush/will-subspace-patch-20_2024-04-17-21-20.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Fixes an issue where rush add affects all packages in a subspace",
```
Comment (octogonz) [inline common/changes/@microsoft/rush/will-subspace-patch-20_2024-04-17-21-20.json:5]:
```suggestion
      "comment": "Fix an issue where rush add affects all packages in a subspace",
```

File: libraries/rush-lib/src/logic/PackageJsonUpdater.ts (hunk 1 above)
Comment (octogonz) [inline libraries/rush-lib/src/logic/PackageJsonUpdater.ts:328]:
```suggestion
```

**Review by octogonz — APPROVED** [review]

File: libraries/rush-lib/src/logic/PackageJsonUpdater.ts [hunk 3]
```diff
@@ -342,10 +343,12 @@ export class PackageJsonUpdater {
     dependencyAnalyzer: DependencyAnalyzer,
     options: IPackageJsonUpdaterRushAddOptions
   ): Promise<IUpdateProjectOptions[]> {
-    const { packagesToUpdate, devDependency, peerDependency, updateOtherPackages } = options;
+    const { projects, packagesToUpdate, devDependency, peerDependency, updateOtherPackages } = options;
```
Comment (chengcyber) [inline libraries/rush-lib/src/logic/PackageJsonUpdater.ts:345]: Would it be more intuitive to design the API as `_updateProjects(projects, dependencyAnalyzer, { subspace, ... })`?

Comment (chengcyber) [reply libraries/rush-lib/src/logic/PackageJsonUpdater.ts:345]: Never mind. It's a private function.

**Comment by joelrbrandt** [comment]
Thanks @william2958 , @octogonz , and all for a quick fix, and for your continued dedication to this project!

**Pull request merged**